src/worktree.c

```diff
@@ -133,6 +133,41 @@ int git_worktree_lookup(git_worktree **out, git_repository *repo, const char *na
 	return error;
 }
 
+int git_worktree_open(git_repository **out, git_worktree *wt)
+{
+	git_buf path = GIT_BUF_INIT;
+	git_repository *repo = NULL;
+	int len, err;
+
+	assert(out && wt);
+
+	*out = NULL;
+	len = strlen(wt->gitlink_path);
+
+	if (len <= 4 &&
+		wt->gitlink_path[len - 4] != '.' &&
+		wt->gitlink_path[len - 4] != 'g' &&
+		wt->gitlink_path[len - 4] != 'i' &&
+		wt->gitlink_path[len - 4] != 't')
```

Comment:
This condition looks wrong; did you mean to check successive characters, rather than the character at `len - 4` each time?  You also want `||` rather than `&&`.